packages/core/crates/core-strategy/src/aggregating.rs

```diff
@@ -0,0 +1,543 @@
+}
+
+impl<Db: HoprCoreEthereumDbActions, T, U> Display for AggregatingStrategy<Db, T, U> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", Strategy::Aggregating(Default::default()))
+    }
+}
+
+impl<Db: HoprCoreEthereumDbActions, T, U> AggregatingStrategy<Db, T, U> {
+    pub fn new(
+        cfg: AggregatingStrategyConfig,
+        db: Arc<RwLock<Db>>,
+        tx_sender: TransactionSender,
+        ticket_aggregator: TicketAggregationActions<T, U>,
+    ) -> Self {
+        Self {
+            cfg,
+            db,
+            tx_sender,
+            ticket_aggregator: Arc::new(Mutex::new(ticket_aggregator)),
+        }
+    }
+}
+
+#[async_trait(? Send)]
+impl<Db: HoprCoreEthereumDbActions + 'static, T, U> SingularStrategy for AggregatingStrategy<Db, T, U> {
+    async fn on_acknowledged_ticket(&self, ack: &AcknowledgedTicket) -> crate::errors::Result<()> {
+        let channel_id = ack.ticket.channel_id;
+
+        let channel = match self.db.read().await.get_channel(&channel_id).await? {
+            Some(channel) => channel,
+            None => {
+                error!("{self} strategy: encountered {ack} in a non-existing channel!");
+                return Err(ChannelDoesNotExist.into());
+            }
+        };
+
+        let acks_in_channel = self
+            .db
+            .read()
+            .await
+            .get_acknowledged_tickets(Some(channel.clone()))
+            .await?
+            .iter()
+            .filter(|ack| ack.status == AcknowledgedTicketStatus::Untouched)
+            .count() as u32;
+
+        if acks_in_channel < self.cfg.aggregation_threshold {
+            debug!(
+                "{self} strategy: {channel} has {acks_in_channel} < {} ack tickets, not aggregating yet",
+                self.cfg.aggregation_threshold
+            );
+            return Ok(());
+        }
+
+        let ticket_aggregation_awaiter = match self
+            .ticket_aggregator
+            .clone()
+            .lock()
+            .await
+            .aggregate_tickets(&channel_id)
+        {
+            Ok(mut awaiter) => awaiter.consume_and_wait(self.cfg.aggregation_timeout).await,
+            Err(e) => {
+                warn!("{self} could not aggregate tickets due to {e}");
+                return Err(crate::errors::StrategyError::Other(e.to_string()));
+            }
+        };
+
+        if let Err(e) = ticket_aggregation_awaiter {
+            error!("{self} aggregation protocol failed due to {}", e.to_string());
+            return Err(crate::errors::StrategyError::Other("ticket aggregation failed".into()));
+        }
+
+        if self.cfg.redeem_after_aggregation {
+            match redeem_tickets_in_channel(self.db.clone(), &channel, true, self.tx_sender.clone()).await {
+                Ok(tx_result) => {
+                    error!("redeeming tickets");
+                    for result in futures::future::join_all(tx_result).await {
```

Comment:
Why await the redeeming to get confirmed?

Reply:
Could this be spawned locally instead of the join_all? How should the results be processed, if not awaited?

Reply:
This would require a dedicated loop to unwrap all those futures.

Reply:
Exactly. Let's just await the one-shot event for now